### DISCUSSION TITLE
Nginx Proxy config for realtime

### DISCUSSION THREAD

**Discussion by jonathan-martz** [pocketbase/pocketbase#5122] [Q&A]
Hi iam currently working on a little project where i want to use the realtime api for updating a card.

My Project has on Instance running from Pocketbase behind a nginx proxy.
I used the Config from https://pocketbase.io/docs/going-to-production/, but there i getting a CORS error for realtime requests.
CORS-Anfrage schlug fehl). Statuscode: (null).
When iam opening the url with my browser it downloads a file.

Is there anything missing in the production example ?
Realtime doesnt work, anything else is working.

`                                                                                                     
server {
    server_name admin.domain.me;

    location / {
        # check http://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive
        proxy_set_header Connection '';
        proxy_http_version 1.1;
        proxy_read_timeout 360s;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        proxy_pass http://127.0.0.1:8090/;
    }

}
server {
    if ($host = admin.domain.me) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    listen 80;
    server_name admin.domain.me;
    return 404; # managed by Certbot
}
`

**Comment by ganigeorgiev** [comment]
The example config from the docs works when testing locally for me.

>  but there i getting a CORS error for realtime requests.

PocketBase allows all origins (`*`) by default (unless explicitly specified otherwise with the `--origins` flag). Note that if the browser returns 404 or some other error for the OPTIONS/preflight request it could be reported as CORS error. Double check that you are using the correct PocketBase url when initializing the SDK and that the nginx https redirect is not conflicting.

You could also try to open the browser inspect tool, navigate to the Network tab and paste/inspect the failed request(s).

**Comment by jonathan-martz** [comment]
The JS SDK is using the *url: 'https://admin.domain.me'* ,
i also tried:

https://stackoverflow.com/questions/75360904/cant-get-pocketbase-running-behind-ngnix-as-a-reverse-proxy
/ add the end of the url

https://github.com/pocketbase/pocketbase/issues/488
proxy_buffering off;

I added a location in the nginx conf for *api/realtime* which has a proxy pass inside and the cors header.
Just returns a 404.
